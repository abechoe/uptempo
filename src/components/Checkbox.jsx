import Radium from 'radium';
import React from 'react';

class Checkbox extends React.Component {
  constructor(props) {
    super(props);

    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    this.props.onChange({[this.props.name]: event.target.checked})
  }

  render() {
    return (
      <div>
        <label htmlFor={this.props.id} style={styles.base}>{this.props.title}</label>
        <input 
          type="checkbox"
          id={this.props.id}
          name={this.props.name}
          onChange={this.handleInputChange}
          style={[styles.base, styles.input]}
        />
      </div>
    );
  }
}

const styles = {
  base: {
    display: 'inline-block',
    marginBottom: '10px',
  },

  input: {
    marginLeft: '5px',
  },
}

export default Radium(Checkbox);