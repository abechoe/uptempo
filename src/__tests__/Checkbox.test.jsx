import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from '../components/Checkbox';

describe('Checkbox component tests', () => {
  it('renders a Checkbox component', () => {
    const wrapper = shallow(<Checkbox id="recurs" name="recurs" />);
    expect(wrapper).toMatchSnapshot();
  })
});