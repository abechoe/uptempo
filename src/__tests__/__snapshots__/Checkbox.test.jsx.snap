// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component tests renders a Checkbox component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    id="recurs"
    name="recurs"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          data-radium={true}
          htmlFor="recurs"
          style={
            Object {
              "display": "inline-block",
              "marginBottom": "10px",
            }
          }
        />,
        <input
          data-radium={true}
          id="recurs"
          name="recurs"
          onChange={[Function]}
          style={
            Object {
              "display": "inline-block",
              "marginBottom": "10px",
              "marginLeft": "5px",
            }
          }
          type="checkbox"
        />,
      ],
      "data-radium": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "data-radium": true,
          "htmlFor": "recurs",
          "style": Object {
            "display": "inline-block",
            "marginBottom": "10px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": ".1",
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "data-radium": true,
          "id": "recurs",
          "name": "recurs",
          "onChange": [Function],
          "style": Object {
            "display": "inline-block",
            "marginBottom": "10px",
            "marginLeft": "5px",
          },
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            data-radium={true}
            htmlFor="recurs"
            style={
              Object {
                "display": "inline-block",
                "marginBottom": "10px",
              }
            }
          />,
          <input
            data-radium={true}
            id="recurs"
            name="recurs"
            onChange={[Function]}
            style={
              Object {
                "display": "inline-block",
                "marginBottom": "10px",
                "marginLeft": "5px",
              }
            }
            type="checkbox"
          />,
        ],
        "data-radium": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "data-radium": true,
            "htmlFor": "recurs",
            "style": Object {
              "display": "inline-block",
              "marginBottom": "10px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "data-radium": true,
            "id": "recurs",
            "name": "recurs",
            "onChange": [Function],
            "style": Object {
              "display": "inline-block",
              "marginBottom": "10px",
              "marginLeft": "5px",
            },
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
